### PR TITLE
Expose collection attributes from `Inspect` trait

### PR DESCRIPTION
**Pull request by dastansam**
# Description

- What does this PR do?

While working with `pallet_nfts` through `nonfungibles_v2` traits `Inspect, Mutate`, I found out that once you have set the collection attribute with `<Nfts as Mutate>::set_collection_attribute()`, it's not possible to read it with `<Nfts as Inspect>::collection_attribute()` since they use different `namespace` values. When setting the attribute, `AttributeNamespace::Pallet` is used, while `AttributeNamespace::CollectionOwner` is used when reading.

more context: https://github.com/freeverseio/laos/issues/7#issuecomment-1766137370

This PR makes `item` an optional parameter in `Inspect::system_attribute()`, to be able to read collection attributes.

- Why are these changes needed?

To be able to read collection level attributes when reading attributes of the collection. It will be possible to read collection attributes by passing `None` for `item`

- How were these changes implemented and what do they affect?

`NftsApi` is also affected and `NftsApi::system_attribute()` now accepts optional `item` parameter. 

## Breaking change

Because of the change in the `NftsApi::system_attribute()` method's `item` param, parachains who integrated the `NftsApi` need to update their API code and frontend integrations accordingly. AssetHubs are unaffected since the NftsApi wasn't released on those parachains yet.

### CODE REVIEW

**Comment by dastansam** [comment]
hey @jsidorenko 

as we [discussed](https://github.com/freeverseio/laos/issues/7#issuecomment-1766146416) earlier, I am submitting this PR. Do you think I should add tests for this? I don't see any tests for trait implementations, so just want to confirm.

File: substrate/frame/support/src/traits/tokens/nonfungible_v2.rs [hunk 1]
```diff
@@ -60,10 +60,12 @@ pub trait Inspect<AccountId> {
...
 	}
 
-	/// Returns the system attribute value of `item` corresponding to `key`.
+	/// Returns the system attribute value of `item` of `collection` corresponding to `key` if
+	/// `item` is `Some`. Otherwise, returns the system attribute value of `collection`
+	/// corresponding to `key`.
 	///
 	/// By default this is `None`; no attributes are defined.
-	fn system_attribute(_item: &Self::ItemId, _key: &[u8]) -> Option<Vec<u8>> {
+	fn system_attribute(_item: Option<&Self::ItemId>, _key: &[u8]) -> Option<Vec<u8>> {
```
Comment (jsidorenko) [inline substrate/frame/support/src/traits/tokens/nonfungible_v2.rs:68]: This file represents the traits to work with a particular item and not a collection, which means you don't need to make the item param optional

File: substrate/frame/support/src/traits/tokens/nonfungible_v2.rs [hunk 2]
```diff
@@ -211,7 +218,7 @@ impl<
 	fn custom_attribute(account: &AccountId, item: &Self::ItemId, key: &[u8]) -> Option<Vec<u8>> {
 		<F as nonfungibles::Inspect<AccountId>>::custom_attribute(account, &A::get(), item, key)
 	}
-	fn system_attribute(item: &Self::ItemId, key: &[u8]) -> Option<Vec<u8>> {
+	fn system_attribute(item: Option<&Self::ItemId>, key: &[u8]) -> Option<Vec<u8>> {
 		<F as nonfungibles::Inspect<AccountId>>::system_attribute(&A::get(), item, key)
 	}
```
Comment (jsidorenko) [inline substrate/frame/support/src/traits/tokens/nonfungible_v2.rs:223]:
```suggestion
	fn system_attribute(item: &Self::ItemId, key: &[u8]) -> Option<Vec<u8>> {
		<F as nonfungibles::Inspect<AccountId>>::system_attribute(&A::get(), Some(item), key)
	}
```

File: substrate/frame/support/src/traits/tokens/nonfungible_v2.rs [hunk 3]
```diff
@@ -229,7 +236,10 @@ impl<
 			key,
 		)
 	}
-	fn typed_system_attribute<K: Encode, V: Decode>(item: &Self::ItemId, key: &K) -> Option<V> {
+	fn typed_system_attribute<K: Encode, V: Decode>(
+		item: Option<&Self::ItemId>,
+		key: &K,
+	) -> Option<V> {
 		<F as nonfungibles::Inspect<AccountId>>::typed_system_attribute(&A::get(), item, key)
 	}
```
Comment (jsidorenko) [inline substrate/frame/support/src/traits/tokens/nonfungible_v2.rs:244]:
```suggestion
	fn typed_system_attribute<K: Encode, V: Decode>(item: &Self::ItemId, key: &K) -> Option<V> {
		<F as nonfungibles::Inspect<AccountId>>::typed_system_attribute(&A::get(), Some(item), key)
	}
```

**Comment by jsidorenko** [comment]
In regards to tests, here is the way to test your changes:
1) `use frame_support::{traits::{tokens::nonfungibles_v2::{Create, Destroy, Inspect, Mutate}}};`
2) ```
		assert_ok!(<Nfts as Mutate<<Test as SystemConfig>::AccountId, ItemConfig>>::set_collection_attribute(
			&0,
			&[1],
			&[2],
		));
		assert_eq!(
			attributes(0),
			vec![
				(None, AttributeNamespace::Pallet, bvec![1], bvec![2]),
			]
		);
		assert_eq!(<Nfts as Inspect<<Test as SystemConfig>::AccountId>>::system_attribute(
			&0,
			None,
			&[1],
		), Some(vec![2]));

```

**Comment by dastansam** [comment]
hey @jsidorenko 

I added tests and I think it's ready for review

**Review by jsidorenko — APPROVED** [review]
Nice!

**Review by seadanda — APPROVED** [review]

**Review by gilescope — APPROVED** [review]

**Review by liamaharon — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-4-0/5152/1

**Pull request merged**